lisp/core.slosh

```diff
@@ -435,12 +435,12 @@ Section: sequence
 
 Example:
 (assert-equal '(2 3) (rest '(1 2 3)))
-(assert-equal '(2 3) (rest [1 2 3]))
+(assert-equal [2 3] (rest [1 2 3]))
 (assert-equal nil (rest '(1)))
-(assert-equal nil (rest [1]))
+(assert-equal [] (rest [1]))
 (assert-equal nil (rest '()))
 (assert-equal nil (rest nil))
-(assert-equal nil (rest []))
+(assert-equal [] (rest []))
```

Comment:
2 example of new behavior for list like data structures with vectors.

Reply:
I think it sticking to the underlying type (vec in this case) and is expected.  We will need to do a review of all this after some dust settles.